### DISCUSSION TITLE
Change build.md to Contributing.md

### DISCUSSION THREAD

**Discussion by NovaliX-Dev** [VSCodium/vscodium#1473] [Ideas]
There is this `build.md` file that looks exactly like a contributing file.

I suggest to rename it to make it more clear, because it's not only about building like it's name suggests.

What are your opinions on this ?

**Comment by daiyam** [comment]
It would nee to be improved.
Looking at the examples found at https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors, https://github.com/rails/rails/blob/main/CONTRIBUTING.md seems to be good start.

**Comment by NovaliX-Dev** [comment]
First i think we should reorganize what's already in the file.

I have many ideas:
- Separate the file in 4 categories : Dictionary (For all the things we should know about), Setup environment (includes installing all dependencies), Build and Contribute (Containing the patch process)
- Add scripts to automate the process in docker (a build one which manage the exports and a setup one which manage the rest) for both architectures
- In the `Build script` section, the Linux and MacOS script is the same, so i think it's more logical to indicate the build script for both of them instead of separated
- Still in the same section, remove the bash script from Windows since powershell is required to be installed
- Make the `Patching process` more organized, especially because the "debugging" (the part about what to do when the patch fails) is in the middle of the main process. I think it should be separated for better readability
- Explain which script does what to clear up any doubt. It can be added instead to the top of those scripts as comments though.

Also removing the `docs` folder (since it only contains this file) and move the build at the root of the repo is a good idea in my opinion.

What do you think about them ?
